### PR TITLE
Adding GCP replica to smoketest

### PR DESCRIPTION
**Pull request by Bento007**
* Moved common setup code to setUpClass
* Added support for testing additional replicas
* Added GSBlobstore for GS checkout verification.

### Test plan
Ran smoke test locally against the dev deployment

### CODE REVIEW

File: tests/test_smoketest.py [hunk 1]
```diff
@@ -207,6 +214,19 @@ def test_smoketest(self):
             self.assertEquals(bundle_uuid, notification['match']['bundle_uuid'])
             self.assertEquals(bundle_version, notification['match']['bundle_version'])
 
+    def test_smoketest(self):
+        for replica in self.replicas:
+            with self.subTest(replica):
+                self.smoketest(replica)
+
+    def get_blobstore(self, replica):
```
Comment (ttung) [inline tests/test_smoketest.py]: I'd type the replica parameter.

File: tests/test_smoketest.py [hunk 2]
```diff
@@ -52,9 +53,12 @@ def run_for_json(command, **kwargs):
...
 class Smoketest(unittest.TestCase):
-    replicas = "aws", "gcp"
+    replicas = {
+        "aws",
+        "gcp"
+    }
     notification_bucket = os.environ['DSS_S3_BUCKET_TEST']
-    checkout_bucket = os.environ['DSS_S3_CHECKOUT_BUCKET']
+    checkout_bucket = {'aws': os.environ['DSS_S3_CHECKOUT_BUCKET'], 'gcp': os.environ['DSS_GS_CHECKOUT_BUCKET']}
     test_bucket = os.environ['DSS_S3_BUCKET_TEST']
```
Comment (ttung) [inline tests/test_smoketest.py]: I think this needs to be parameterized too. 

Might be worth creating a replica to parameterization values dict.

File: tests/test_smoketest.py [hunk 3]
```diff
@@ -110,89 +113,93 @@ def test_smoketest(self):
...
-        run(f"{venv_bin}hca dss download --replica gcp --bundle-uuid {bundle_uuid}")
+            # Download bundle from other replica
+            #
+            run(f"{self.venv_bin}hca dss download --replica {replica} --bundle-uuid {bundle_uuid}")
 
-        # Run a CLI search against the two replicas
-        #
         for replica in self.replicas:
-            run(f"{venv_bin}hca dss post-search --es-query='{{}}' --replica {replica} > /dev/null")
+            # Run a CLI search against the replicas
```
Comment (ttung) [inline tests/test_smoketest.py]: As the number of replicas potentially increases, this may be untenable.

**Comment by Bento007** [comment]
Ready for review

File: tests/test_smoketest.py [hunk 4]
```diff
@@ -52,13 +54,23 @@ def run_for_json(command, **kwargs):
 
 @testmode.integration
 class Smoketest(unittest.TestCase):
-    replicas = "aws", "gcp"
+    params = [
```
Comment (ttung) [inline tests/test_smoketest.py]: I would do this instead:

```
REPLICAS = {
  Replica.AWS: {
    'checkout_bucket_envvar': "DSS_S3_CHECKOUT_BUCKET",
    'test_bucket_envvar': "DSS_S3_BUCKET_TEST",
  },
  Replica.GCP: {
    'checkout_bucket_envvar': "DSS_GS_CHECKOUT_BUCKET",
    'test_bucket_envvar': "DSS_GS_BUCKET_TEST",
  },
}
```

**Comment by ttung** [comment]
My comment about the number of replicas making things untenable is really a question directed at _all_ the code that checks / downloads from the other replicas, not just search.  I don't know if you need to special case search specifically, or even at all now.  It was a general comment for something we should pay attention to as we expand the # of replicas.

I think the right solution might be to continue checking each <cloud-provider> <-> <cloud-provider> combination.

File: tests/test_smoketest.py (hunk 4 above)
Comment (Bento007) [reply tests/test_smoketest.py]: Changes it to use Replica, but I kept it as a list.

**Pull request merged**